aws-observability/aws-otel-js-instrumentation
# feat: [.NET and JS Feature Parity] Support for new AWS Resources in JS SDK

*Description of changes:*

Adding support for new AWS resources in JS SDK. Part of an ongoing project to increase the ADOT SDK support in Node and .NET. 

Changes in this PR support the exact same features as the Python version: 
https://github.com/aws-observability/aws-otel-python-instrumentation/pull/265

Manual Testing:
<img width="1236" alt="Screenshot 2024-11-15 at 1 36 01 PM" src="https://github.com/user-attachments/assets/7480f865-2927-4376-91b5-76048b0bcdf4">
<img width="1234" alt="Screenshot 2024-11-15 at 2 07 05 PM" src="https://github.com/user-attachments/assets/1a5d00b4-9ac8-45e9-b51d-02d7c669f5f1">
<img width="939" alt="Screenshot 2024-11-15 at 2 26 10 PM" src="https://github.com/user-attachments/assets/120892ec-03cf-4b70-80c7-167369cd9e27">
<img width="996" alt="Screenshot 2024-11-18 at 9 47 33 AM" src="https://github.com/user-attachments/assets/08019f4c-507c-46cf-baa0-db77a4175d02">
<img width="934" alt="Screenshot 2024-11-18 at 9 57 22 AM" src="https://github.com/user-attachments/assets/3fb79fa9-689f-4361-b671-79e23c68193d">
<img width="852" alt="Screenshot 2024-11-18 at 10 00 34 AM" src="https://github.com/user-attachments/assets/bde63f88-cdbc-43c9-aa16-068790556ebe">

Unit Tests for Instrumentation Patches and Metric Attribute Generators:

<img width="1070" alt="image" src="https://github.com/user-attachments/assets/df01814c-49fb-4561-879b-c88cedb35e5c">

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
